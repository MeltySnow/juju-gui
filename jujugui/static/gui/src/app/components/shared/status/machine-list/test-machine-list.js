/* Copyright (C) 2018 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const StatusMachineList = require('./machine-list');

describe('StatusMachineList', () => {
  let machines;

  const renderComponent = (options = {}) => enzyme.shallow(
    <StatusMachineList
      changeState={options.changeState || sinon.stub()}
      generateMachineClickState={
        options.generateMachineClickState || sinon.stub().returns({ app: 'state' })}
      generatePath={options.generatePath || sinon.stub()}
      machines={options.machines || machines}
      statusFilter={options.statusFilter} />
  );

  beforeEach(() => {
    machines = [{
      agent_state: 'pending',
      agent_state_info: '',
      displayName: '1',
      id: 'm1',
      instance_id: 'machine-1',
      public_address: '1.2.3.6',
      series: 'zesty'
    }, {
      agent_state: 'started',
      agent_state_info: 'yes, I am started',
      displayName: '2',
      id: 'm2',
      instance_id: 'machine-2',
      public_address: '1.2.3.7',
      series: 'trusty'
    }, {
      // Uncommitted machines are excluded.
      agent_state: '',
      agent_state_info: '',
      displayName: '3',
      id: 'new1',
      instance_id: '',
      public_address: '',
      series: 'trusty'
    }];
  });

  it('renders', () => {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });
});
