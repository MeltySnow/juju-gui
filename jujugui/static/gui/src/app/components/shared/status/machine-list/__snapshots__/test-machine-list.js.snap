// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusMachineList renders 1`] = `
<StatusTable
  headers={
    Array [
      Object {
        "columnSize": 1,
        "content": "Machine",
      },
      Object {
        "columnSize": 2,
        "content": "State",
      },
      Object {
        "columnSize": 2,
        "content": "DNS",
      },
      Object {
        "columnSize": 3,
        "content": "Instance ID",
      },
      Object {
        "columnSize": 1,
        "content": "Series",
      },
      Object {
        "columnSize": 3,
        "content": "Message",
      },
    ]
  }
  rows={
    Array [
      Object {
        "classes": Array [
          "status-table__row--pending",
        ],
        "clickURL": "http://example.com",
        "columns": Array [
          Object {
            "columnSize": 1,
            "content": "1",
          },
          Object {
            "columnSize": 2,
            "content": <StatusLabel
              status="pending"
            />,
          },
          Object {
            "columnSize": 2,
            "content": "1.2.3.6",
          },
          Object {
            "columnSize": 3,
            "content": "machine-1",
          },
          Object {
            "columnSize": 1,
            "content": "zesty",
          },
          Object {
            "columnSize": 3,
            "content": "",
          },
        ],
        "extraData": "pending",
        "key": "m1",
        "onClick": [Function],
      },
      Object {
        "classes": Array [
          "status-table__row--ok",
        ],
        "clickURL": "http://example.com",
        "columns": Array [
          Object {
            "columnSize": 1,
            "content": "2",
          },
          Object {
            "columnSize": 2,
            "content": <StatusLabel
              status="started"
            />,
          },
          Object {
            "columnSize": 2,
            "content": "1.2.3.7",
          },
          Object {
            "columnSize": 3,
            "content": "machine-2",
          },
          Object {
            "columnSize": 1,
            "content": "trusty",
          },
          Object {
            "columnSize": 3,
            "content": "yes, I am started",
          },
        ],
        "extraData": "ok",
        "key": "m2",
        "onClick": [Function],
      },
      Object {
        "classes": Array [
          "",
        ],
        "clickURL": "http://example.com",
        "columns": Array [
          Object {
            "columnSize": 1,
            "content": "3",
          },
          Object {
            "columnSize": 2,
            "content": null,
          },
          Object {
            "columnSize": 2,
            "content": "",
          },
          Object {
            "columnSize": 3,
            "content": "",
          },
          Object {
            "columnSize": 1,
            "content": "trusty",
          },
          Object {
            "columnSize": 3,
            "content": "",
          },
        ],
        "extraData": "ok",
        "key": "new1",
        "onClick": [Function],
      },
    ]
  }
/>
`;
